python-flask/app.py

```diff
@@ -20,17 +20,14 @@
 # with Crate these terms will supersede the license and you may use the
 # software solely pursuant to the terms of the relevant commercial agreement.
 
-import json
 import uuid
 import base64
 import hashlib
 
 from datetime import datetime
 from tempfile import TemporaryFile
 
-from urllib.request import urlopen, Request
-from urllib.error import HTTPError, URLError
-
+import werkzeug.exceptions
```

Comment:
Do we need to add `werkzeug` as a dependency in `pyproject.toml`? It works right now, I guess because `werkzeug` is an indirect dependency?

Reply:
Yes. Flask is based on Werkzeug, and as far as I can see, there is no Flask-specific exception wrapper around Werkzeug's `BadRequest` class.

Reply:
With ce352b36c, I've now implemented a much better solution, by overriding Flask's `on_json_loading_failed` request method, as suggested at [^1].

[^1]: https://stackoverflow.com/a/39451091